### PR TITLE
PP-602/RelayClient: _validateSmartWallet

### PR DESCRIPTION
**Pull request by franciscotobar**
## What

- Add relay client
- Re implement _validateSmartWallet

## Refs

- [PP-602](https://rsklabs.atlassian.net/browse/PP-602)
- https://github.com/rsksmart/rif-relay-contracts/pulls

### CODE REVIEW

**Comment by jurajpiar** [comment]
rebased on latest `feature/ethers/dependencies`

**Review by jurajpiar — APPROVED** [review]
LG

**Comment by antomor** [comment]
rebased on latest `feature/ethers/dependencies`

File: src/RelayClient.ts [hunk 1]
```diff
@@ -125,6 +126,15 @@ class RelayClient extends EnvelopingEventEmitter {
 
     return BigNumber.from(gasPrice.toFixed());
   };
+  public async isSmartWalletOwner(smartWalletAddress: string, owner:string): Promise<boolean>{
+    const iForwarder = IForwarder__factory.connect(smartWalletAddress, this._provider);
+    if(await iForwarder.getOwner()  !== utils.solidityKeccak256(['address'], [owner])){
+      return false;
+    }
+
+    return true;
```
Comment (franciscotobar) [inline src/RelayClient.ts:135]: Over here Im not sure if I should return a flag or throw an error. @rsksmart/payments

Comment (AndresQuijano) [reply src/RelayClient.ts:135]: The goal of the method is to check if it is the owner or not. I think your current approach is correct. If an error is required to be thrown, the caller  should throw it.

**Review by AndresQuijano — APPROVED** [review]

File: src/RelayClient.ts (hunk 1 above)
Comment (antomor) [reply src/RelayClient.ts:135]: 👆🏽

**Pull request merged**